lib/message-queue.js

```diff
@@ -85,13 +84,31 @@ function MessageQueue() {
 
   //region Private Helper Methods
 
-  function _startPolling() {
+  async function _startPolling() {
     if (!_pollingIntervalId) {
+      // Temporarily assigning polling interval to prevent multiple polls starting while index is being created
+      _pollingIntervalId = -1;
+
+      if (self.createIndex) {
+        await _createQueueIndex();
+      }
+
       // Try and find work at least once every pollingInterval
       _pollingIntervalId = setInterval(_poll, self.pollingInterval);
     }
   }
 
+  async function _createQueueIndex() {
+    const collection = await _getCollection();
+
+    await collection.createIndex({
```

Comment:
Probably want to add the options argument to force the name, unique:false, and sparse:false